File: app/src/main/java/com/expansion/lg/kimaru/expansion/mzigos/Interview.java
package com.expansion.lg.kimaru.expansion.mzigos;

import com.expansion.lg.kimaru.expansion.activity.MainActivity;
import com.expansion.lg.kimaru.expansion.tables.RegistrationTable;

/**
 * Created by kimaru on 3/2/17.
 */

public class Interview {

    private Integer motivation, community, mentality, selling, health, investment;
    private Integer interpersonal, commitment, total, addedBy, synced, selected, readAndInterpret,
            interviewerMotivationAssessment, interviewerAgeAssessment, interviewerResidenyAssessment,
            interviewerBracAssessment, interviewerAbilityToReadAssessment, interviewerQualifyAssessment;

    private String comment, id, applicant, recruitment, country;
    Long dateAdded;
    private boolean isRead, isImportant, canJoin;
    int color = -1;
    String picture =  "";

    public Interview(){

    }

    public Interview(String id, String applicant, String recruitment, Integer motivation, Integer community,
                     Integer mentality, Integer selling, Integer health, Integer investment,
                     Integer interpersonal, Integer commitment, Integer selected, Integer addedBy,
                     Long dateAdded, Integer synced, String comment, boolean canJoin, String country,
                     Integer readAndInterpret, Integer interviewerMotivationAssessment, Integer interviewerAgeAssessment, Integer interviewerResidenyAssessment,
                     Integer interviewerBracAssessment, Integer interviewerAbilityToReadAssessment, Integer interviewerQualifyAssessment) {
        this.applicant = applicant;
        this.recruitment = recruitment;
        this.motivation = motivation;
        this.community = community;
        this.mentality = mentality;
        this.selling = selling;
        this.health = health;
        this.investment = investment;
        this.interpersonal = interpersonal;
        this.commitment = commitment;
        this.selected = selected;
        this.addedBy = addedBy;
        this.dateAdded = dateAdded;
        this.synced = synced;
        this.comment = comment;
        this.country = country;
        this.id = id;
        this.canJoin = canJoin;
        this.interviewerMotivationAssessment=interviewerMotivationAssessment;
        this.interviewerAgeAssessment=interviewerAgeAssessment;
        this.interviewerResidenyAssessment=interviewerResidenyAssessment;
        this.interviewerBracAssessment=interviewerBracAssessment;
        this.interviewerAbilityToReadAssessment=interviewerAbilityToReadAssessment;
        this.interviewerQualifyAssessment=interviewerQualifyAssessment;
        this.readAndInterpret = readAndInterpret;
        this.total = (motivation + community + mentality + selling + health + investment + interpersonal + commitment);
    }


    // Get Methods


    public Integer getInterviewerMotivationAssessment() {
        return interviewerMotivationAssessment;
    }

    public Integer getInterviewerAgeAssessment() {
        return interviewerAgeAssessment;
    }

    public Integer getInterviewerResidenyAssessment() {
        return interviewerResidenyAssessment;
    }

    public Integer getInterviewerBracAssessment() {
        return interviewerBracAssessment;
    }

    public Integer getInterviewerAbilityToReadAssessment() {
        return interviewerAbilityToReadAssessment;
    }

    public Integer getInterviewerQualifyAssessment() {
        return interviewerQualifyAssessment;
    }

    public String getApplicant() {
        return applicant;
    }

    public String getRecruitment() {
        return recruitment;
    }

    public Integer getMotivation() {
        return motivation;
    }

    public Integer getCommunity() {
        return community;
    }

    public Integer getMentality() {
        return mentality;
    }

    public Integer getSelling() {
        return selling;
    }

    public Integer getHealth() {
        return health;
    }

    public Integer getInvestment() {
        return investment;
    }

    public Integer getInterpersonal() {
        return interpersonal;
    }

    public Integer getCommitment() {
        return commitment;
    }

    public Integer getSelected() {
        return selected;
    }

    public Integer getAddedBy() {
        return addedBy;
    }

    public Long getDateAdded() {
        return dateAdded;
    }

    public Integer getSynced() {
        return synced;
    }

    public Integer getTotal() {
        if (!(this.total==null)){
            return this.total;
        }else{
            return this.calculateTotal();
        }

    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public boolean isCanJoin() {
        return canJoin;
    }

    public void setCanJoin(boolean canJoin) {
        this.canJoin = canJoin;
    }

    public String getComment() {
        return comment;
    }

    public String getId() {
        return id;
    }

    public boolean isImportant() {
        return isImportant;
    }

    public boolean isRead() {
        return isRead;
    }

    public int getColor() {
        return color;
    }

    public String getPicture() {
        return picture;
    }

    private Integer calculateTotal(){
        Integer interviewTotal = 0;
        Integer motivation = this.motivation == null ? 0: this.motivation;
        Integer community = this.community == null ? 0: this.community;
        Integer mentality = this.mentality == null ? 0: this.mentality;
        Integer selling  = this.selling  == null ? 0: this.selling;
        Integer health  =this.health  == null ? 0: this.health;
        Integer investment = this.investment == null ? 0: this.investment;
        Integer interpersonal  =this.interpersonal  == null ? 0: this.interpersonal;
        Integer commitment = this.commitment == null ? 0: this.commitment;
        interviewTotal = motivation +community+mentality+selling+health+investment+interpersonal+commitment;
        return interviewTotal;
    }

    public Integer getReadAndInterpret() {
        return readAndInterpret;
    }

    //Set Methods


    public void setRecruitment(String recruitment) {
        this.recruitment = recruitment;
    }

    public void setCommitment(Integer commitment) {
        this.commitment = commitment;
    }

    public void setCommunity(Integer community) {
        this.community = community;
    }

    public void setHealth(Integer health) {
        this.health = health;
    }

    public void setInterpersonal(Integer interpersonal) {
        this.interpersonal = interpersonal;
    }

    public void setInvestment(Integer investment) {
        this.investment = investment;
    }

    public void setMentality(Integer mentality) {
        this.mentality = mentality;
    }

    public void setMotivation(Integer motivation) {
        this.motivation = motivation;
    }

    public void setSelected(Integer selected) {
        this.selected = selected;
    }

    public void setSelling(Integer selling) {
        this.selling = selling;
    }


    public void setApplicant(String applicant) {
        this.applicant = applicant;
    }

    public void setSynced(Integer synced) {
        this.synced = synced;
    }

    public void setDateAdded(Long dateAdded) {
        this.dateAdded = dateAdded;
    }

    public void setAddedBy(Integer addedBy) {
        this.addedBy = addedBy;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void setImportant(boolean important) {
        isImportant = important;
    }

    public void setRead(boolean read) {
        isRead = read;
    }

    public void setPicture(String picture) {
        this.picture = picture;
    }

    public void setColor(int color) {
        this.color = color;
    }

    public boolean hasPassed(){
        if (this.commitment > 1 && getTotal() > 24 && canJoin ){
            return true;
        }else{
            return false;
        }
    }

    public void setReadAndInterpret(Integer readAndInterpret) {
        this.readAndInterpret = readAndInterpret;
    }

    public void setInterviewerMotivationAssessment(Integer interviewerMotivationAssessment) {
        this.interviewerMotivationAssessment = interviewerMotivationAssessment;
    }

    public void setInterviewerAgeAssessment(Integer interviewerAgeAssessment) {
        this.interviewerAgeAssessment = interviewerAgeAssessment;
    }

    public void setInterviewerResidenyAssessment(Integer interviewerResidenyAssessment) {
        this.interviewerResidenyAssessment = interviewerResidenyAssessment;
    }

    public void setInterviewerBracAssessment(Integer interviewerBracAssessment) {
        this.interviewerBracAssessment = interviewerBracAssessment;
    }

    public void setInterviewerAbilityToReadAssessment(Integer interviewerAbilityToReadAssessment) {
        this.interviewerAbilityToReadAssessment = interviewerAbilityToReadAssessment;
    }

    public void setInterviewerQualifyAssessment(Integer interviewerQualifyAssessment) {
        this.interviewerQualifyAssessment = interviewerQualifyAssessment;
    }
}

File: app/src/main/java/com/expansion/lg/kimaru/expansion/mzigos/Interview.kt
package com.expansion.lg.kimaru.expansion.mzigos

import com.expansion.lg.kimaru.expansion.activity.MainActivity
import com.expansion.lg.kimaru.expansion.tables.RegistrationTable

/**
 * Created by kimaru on 3/2/17.
 */

class Interview {

    var motivation: Int? = null
    var community: Int? = null
    var mentality: Int? = null
    var selling: Int? = null
    var health: Int? = null
    var investment: Int? = null
    var interpersonal: Int? = null
    var commitment: Int? = null
    internal var total: Int? = null
    var addedBy: Int? = null
    var synced: Int? = null
    var selected: Int? = null
    var readAndInterpret: Int? = null
    // Get Methods


    var interviewerMotivationAssessment: Int? = null
    var interviewerAgeAssessment: Int? = null
    var interviewerResidenyAssessment: Int? = null
    var interviewerBracAssessment: Int? = null
    var interviewerAbilityToReadAssessment: Int? = null
    var interviewerQualifyAssessment: Int? = null
    lateinit var comment: String
    lateinit var id: String
    lateinit var applicant: String
    //Set Methods


    lateinit var recruitment: String
    lateinit var country: String
    var dateAdded: Long? = null
    var isRead: Boolean = false
    var isImportant: Boolean = false
    var isCanJoin: Boolean = false
    var color = -1
    var picture = ""

    constructor() {

    }

    constructor(id: String, applicant: String, recruitment: String, motivation: Int?, community: Int?,
                mentality: Int?, selling: Int?, health: Int?, investment: Int?,
                interpersonal: Int?, commitment: Int?, selected: Int?, addedBy: Int?,
                dateAdded: Long?, synced: Int?, comment: String, canJoin: Boolean, country: String,
                readAndInterpret: Int?, interviewerMotivationAssessment: Int?, interviewerAgeAssessment: Int?, interviewerResidenyAssessment: Int?,
                interviewerBracAssessment: Int?, interviewerAbilityToReadAssessment: Int?, interviewerQualifyAssessment: Int?) {
        this.applicant = applicant
        this.recruitment = recruitment
        this.motivation = motivation
        this.community = community
        this.mentality = mentality
        this.selling = selling
        this.health = health
        this.investment = investment
        this.interpersonal = interpersonal
        this.commitment = commitment
        this.selected = selected
        this.addedBy = addedBy
        this.dateAdded = dateAdded
        this.synced = synced
        this.comment = comment
        this.country = country
        this.id = id
        this.isCanJoin = canJoin
        this.interviewerMotivationAssessment = interviewerMotivationAssessment
        this.interviewerAgeAssessment = interviewerAgeAssessment
        this.interviewerResidenyAssessment = interviewerResidenyAssessment
        this.interviewerBracAssessment = interviewerBracAssessment
        this.interviewerAbilityToReadAssessment = interviewerAbilityToReadAssessment
        this.interviewerQualifyAssessment = interviewerQualifyAssessment
        this.readAndInterpret = readAndInterpret
        this.total = motivation!! + community!! + mentality!! + selling!! + health!! + investment!! + interpersonal!! + commitment!!
    }

    fun getTotal(): Int? {
        return if (this.total != null) {
            this.total
        } else {
            this.calculateTotal()
        }

    }

    private fun calculateTotal(): Int? {
        var interviewTotal: Int? = 0
        val motivation = if (this.motivation == null) 0 else this.motivation
        val community = if (this.community == null) 0 else this.community
        val mentality = if (this.mentality == null) 0 else this.mentality
        val selling = if (this.selling == null) 0 else this.selling
        val health = if (this.health == null) 0 else this.health
        val investment = if (this.investment == null) 0 else this.investment
        val interpersonal = if (this.interpersonal == null) 0 else this.interpersonal
        val commitment = if (this.commitment == null) 0 else this.commitment
        interviewTotal = motivation!! + community!! + mentality!! + selling!! + health!! + investment!! + interpersonal!! + commitment!!
        return interviewTotal
    }

    fun hasPassed(): Boolean {
        return this.commitment!! > 1 && getTotal()!! > 24 && isCanJoin
    }
}
